homeassistant/components/anthropic/config_flow.py

```diff
@@ -128,21 +132,50 @@ async def async_step_init(
     ) -> ConfigFlowResult:
         """Manage the options."""
         options: dict[str, Any] | MappingProxyType[str, Any] = self.config_entry.options
+        errors: dict[str, str] = {}
 
         if user_input is not None:
             if user_input[CONF_RECOMMENDED] == self.last_rendered_recommended:
                 if user_input[CONF_LLM_HASS_API] == "none":
                     user_input.pop(CONF_LLM_HASS_API)
-                return self.async_create_entry(title="", data=user_input)
 
-            # Re-render the options again, now with the recommended options shown/hidden
-            self.last_rendered_recommended = user_input[CONF_RECOMMENDED]
+                if (
+                    user_input.get(
+                        CONF_THINKING_BUDGET_TOKENS, RECOMMENDED_THINKING_BUDGET_TOKENS
+                    )
+                    >= MIN_THINKING_BUDGET
+                    and user_input.get(CONF_CHAT_MODEL, RECOMMENDED_CHAT_MODEL)
+                    not in THINKING_MODELS
+                ):
+                    errors[CONF_THINKING_BUDGET_TOKENS] = (
+                        "model_does_not_support_thinking"
```

Comment:
Should we throw this error or just ignore the value instead?

Reply:
I am not sure. We don't want to create false expectations. There is a text in documentation that it is only applicable to extended thinking models, but not everyone would read it.

Reply:
If we ignore and don't throw an error, then Anthropic API would.
![image](https://github.com/user-attachments/assets/d9093942-45da-4f31-a675-c10ecf89a3ba)

Reply:
Right, but wouldn't it be better if we can just not pass that value, and then just work?

Reply:
Agree. I've made a note on it in the documentation PR.